src/main/scala/no/ndla/searchapi/model/taxonomy/Resource.scala

```diff
@@ -7,4 +7,18 @@
 
 package no.ndla.searchapi.model.taxonomy
 
-case class Resource(id: String, name: String, contentUri: Option[String], path: String)
+abstract class TaxonomyElement(id: String,
+                               name: String,
+                               contentUri: Option[String],
+                               path: String,
+                               metadata: Option[Metadata]) {
+  def getId(): String = id
+  def getName(): String = name
+}
+
+case class TaxSubject(id: String, name: String, contentUri: Option[String], path: String, metadata: Option[Metadata])
+    extends TaxonomyElement(id, name, contentUri, path, metadata)
+case class Resource(id: String, name: String, contentUri: Option[String], path: String, metadata: Option[Metadata])
+    extends TaxonomyElement(id, name, contentUri, path, metadata)
+case class Topic(id: String, name: String, contentUri: Option[String], path: String, metadata: Option[Metadata])
+    extends TaxonomyElement(id, name, contentUri, path, metadata)
```

Comment:
Helt subjektivt syns jeg dette blir penere om det er en trait og vi slipper å definere getters selv.
Da slipper vi også å sende inn parameterene der vi extender.

```suggestion
sealed trait TaxonomyElement {
  val id: String
  val name: String
  val contentUri: Option[String]
  val path: String
  val metadata: Option[Metadata]
}

case class TaxSubject(
    id: String,
    name: String,
    contentUri: Option[String],
    path: String,
    metadata: Option[Metadata]
) extends TaxonomyElement

case class Resource(
    id: String,
    name: String,
    contentUri: Option[String],
    path: String,
    metadata: Option[Metadata]
) extends TaxonomyElement

case class Topic(
    id: String,
    name: String,
    contentUri: Option[String],
    path: String,
    metadata: Option[Metadata]
) extends TaxonomyElement
```

Reply:
Gjerne. Eg kan for lite scala så det var noko sånt eg egentlig ville ha.